src/Wallpaper/Portal.vala

```diff
@@ -11,42 +11,194 @@ public class Wallpaper.Portal : Object {
+
+
+
+
         window.set_child(box);
         window.show();
+
+    }
+
+    public void update_xresources_wallpaper(string wallpaper_url, string key) throws Error {
+         // Define the path to the Xresources file
+        string xresources_file_path = Path.build_filename(Environment.get_home_dir(), ".config", "regolith", "Xresources");
+
+         // Check if the directory exists, create it if not
+        File directory = File.new_for_path(Path.get_dirname(xresources_file_path));
+        if (!directory.query_exists()) {
+            try {
+                directory.make_directory_with_parents();
+                stdout.printf("Directory created: %s\n", directory.get_path());
+            } catch (Error e) {
+                stderr.printf("Error creating directory: %s\n", e.message);
+                return;
+            }
+        }
+
+         // Now proceed to check and create the file
+        File file = File.new_for_path(xresources_file_path);
+
+        // Check if the file exists
+        if (!file.query_exists()) {
+            stdout.printf("Xresources file does not exist: %s\n", xresources_file_path);
+            // Create the file if it doesn't exist
+            try {
+                GLib.FileOutputStream? fs = file.create(FileCreateFlags.NONE);
+                // Close the file stream after creating
+                if (fs != null)
+                    fs.close(null);
+                stdout.printf("Xresources file created: %s\n", xresources_file_path);
+            } catch (Error e) {
+                stderr.printf("Error creating file: %s\n", e.message);
+                return;
+            }
+        } else {
+            stdout.printf("Xresources file exists: %s\n", xresources_file_path);
+        }
+
+        try {
+            // Read existing contents
+            uint8[] contents;
+            string etag_out;
+
+            bool loading_operation = file.load_contents(null, out contents, out etag_out);
+
+            if(!loading_operation) {
+                stderr.printf("Error loading Xresources file contents\n");
+                return;
+            }
+
+            
+            if (contents != null) {
+                // Convert contents from bytes to string
+                string contentsString = (string)contents;
+
+               // Check if key exists for regolith.wallpaper.file and update it, if it doesn't exist create one
+                string line = key + ": " + wallpaper_url + "\n";
+
+                if (contentsString.contains(key)) {
+                    // Replace the existing line
+
+                    int index = contentsString.index_of(key);
+                    int newlineIndex = contentsString.index_of("\n", index);
+
+                    string old_line = contentsString.substring(index, newlineIndex - index + 1);
+
+                    stdout.printf("old_line: %s", old_line);
+
+
+                    contentsString = contentsString.replace( old_line, line);
```

Comment:
This looks good, but my preference would be to comment the old line and set the resource values in a new line. Also delete any existing comments to avoid cluttering up the xresources.